portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -23,13 +24,31 @@
 public class HomeServlet extends HttpServlet {
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("text/html");
-
+        Gson gson = new Gson(); 
+        response.setContentType("application/json");
         UserService userService = UserServiceFactory.getUserService();
+        
+        //The first field is the logged-in user's email, the second field is the 
+        //URL to login/logout depending on current status
+        String[] loginStatus = new String[2];
```

Comment:
Consider making a LoginStatus class with two fields email and url and using Gson in the same way to [convert it to JSON](https://github.com/google/gson/blob/master/UserGuide.md#object-examples). This could improve the code a bit as field names are easier to read and less error prone than array indexes. This will also help the code in `script.js `.

Reply:
Thanks for letting me know, I was wondering also wondering if there is a way to use java class methods on those objects in the `script.js` file? For example, if I wanted to shorten `result.email.length >0` to be its own method in the UserLogin class, would I be able to use that method in JavaScript?

Reply:
Unfortunately the methods you create in Java will not be transferred by JSON to JavaScript. Only the object's field names and their values so you would need to create a duplicate function in the `script.js`.